features/users/user_rename.feature

```diff
@@ -163,3 +163,17 @@ Feature:
       And I should see "Hi, newusername"
     When I follow "Series"
     Then I should see "Best Series by newusername"
+
+    Scenario: Changing the username from a forbidden name to non-forbidden
+      Given I have no users
+        And the following activated user exists
+          | login     | password |
+          | forbidden | secret   |
+        And the user name "forbidden" is on the forbidden list
+      When I am logged in as "forbidden" with password "password"
```

Comment:
I am... deeply sorry to be this person, by why on earth is it letting us log in with a password that doesn't match the one set two lines up? 😬 😭

Reply:
No worries, I am happy to have anyone pry when something looks weird versus the alternative :slightly_smiling_face: . It looks like
1. the step calls `find_or_create_new_user`: https://github.com/otwcode/otwarchive/blob/dba57ea17dbdef10c4cacb1fffdad72af92f4cd6/features/step_definitions/user_steps.rb#L97
2. that function finds the existing user: https://github.com/otwcode/otwarchive/blob/92d7fc675437c76b8f1e33751ac3957224b064f0/features/support/user.rb#L3
3. and "helpfully" (? on the quotation marks) changes the password to whatever we say it is at login: https://github.com/otwcode/otwarchive/blob/92d7fc675437c76b8f1e33751ac3957224b064f0/features/support/user.rb#L12

So I think that's intended behaviour. I've changed this particular test, to avoid confusion later. We might want to change how this works later, but I suspect it will involve some amount of pain to get all the Cucumber specs passing.

Reply:
"Helpfully" indeed. Thanks for investigating! We should probably make a low priority issue to change that at some point.